### PR TITLE
Feature 13

### LINKED ISSUE #13: Compression

### PR DESCRIPTION
**Pull request by Microsamp**
Closes #13

### CODE REVIEW

**Comment by Microsamp** [comment]
Still WIP: The compressor seems not to be called

File: common/src/main/java/de/prokyo/network/common/compression/CompressionUtil.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package de.prokyo.network.common.compression;
+
+import lombok.Getter;
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Decompressor;
+import net.jpountz.lz4.LZ4Factory;
+
```
Comment (Masy) [inline common/src/main/java/de/prokyo/network/common/compression/CompressionUtil.java:7]: Missing javadoc

File: common/src/main/java/de/prokyo/network/common/compression/CompressionUtil.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+public class CompressionUtil {
+
+	@Getter private static CompressionUtil instance;
+
+	private LZ4Factory lz4Factory;
+	private LZ4Compressor lz4Compressor;
+	private LZ4Decompressor lz4Decompressor;
+
```
Comment (Masy) [inline common/src/main/java/de/prokyo/network/common/compression/CompressionUtil.java:18]: Missing javadoc

File: common/src/main/java/de/prokyo/network/common/compression/CompressionUtil.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	/**
+	 * Initializes the singleton instance with given arguments.
+	 *
+	 * @param type The type of the compression algorithm used
+	 */
+	public static void init(CompressionType type) {
+		CompressionUtil.instance = new CompressionUtil(type);
+	}
+
```
Comment (Masy) [inline common/src/main/java/de/prokyo/network/common/compression/CompressionUtil.java:78]: Missing javadoc..?

File: common/src/main/java/de/prokyo/network/common/pipeline/ProkyoDecompressor.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package de.prokyo.network.common.pipeline;
+
+import de.prokyo.network.common.buffer.PacketBuffer;
+import de.prokyo.network.common.compression.CompressionUtil;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageDecoder;
+import java.util.List;
+
```
Comment (Masy) [inline common/src/main/java/de/prokyo/network/common/pipeline/ProkyoDecompressor.java:9]: Missing javadoc

**Review by Masy — APPROVED** [review]

**Pull request merged**